dll/win32/shell32/CShellLink.cpp

```diff
@@ -2789,7 +2794,8 @@ BOOL CShellLink::OnInitDialog(HWND hwndDlg, HWND hwndFocus, LPARAM lParam)
     /* Target path */
     if (m_sPath)
     {
-        WCHAR newpath[2*MAX_PATH] = L"\0";
+        WCHAR newpath[2*MAX_PATH];
+        newpath[0] = UNICODE_NULL;
```

Comment:
Not sure we need to preinitialize the buffer there.
Indeed, either the `StringCchPrintfExW` or the `StringCchCopyExW` calls just below should initialize the buffer to "something", even just with a single NULL-terminator in case the string to be copied is empty.

Reply:
This was existing code, I was just getting rid of the pointless call to memset.

Reply:
All the calls on this buffer are `StringCch...` so it should behave reasonably. All I really care about here is that if everything somehow fails, `SetDlgItemTextW(hwndDlg, IDC_SHORTCUT_TARGET_TEXT, newpath);` gets an empty string.